cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/NodeMapping.java

```diff
@@ -47,47 +49,123 @@ public int iidmNodeForTerminal(CgmesTerminal t, boolean isSwitchEnd, VoltageLeve
+        var fictSwCreationMode = context.config().getCreateFictitiousSwitchesForDisconnectedTerminalsMode();
+        boolean bbsForEveryConnectivityNode = context.config().createBusbarSectionForEveryConnectivityNode();
+        if (!bbsForEveryConnectivityNode && fictSwCreationMode == CgmesImport.FictitiousSwitchesCreationMode.NEVER) {
+            if (isSwitchEnd) {
+                // for switches the iidm node is the one from the connectivity node
+                return cgmes2iidm.get(t.connectivityNode());
+            } else if (!cgmes2iidm.containsKey(t.id())) {
+                // Create internal connections but only if too many terminals on connectivity node
+                createInternalConnectionsIfNeeded(t.connectivityNode(), vl);
+            }
+        } else {
+            boolean connected = t.connected() && equipmentIsConnected;
+            if (!connected && createFictitiousSwitch(fictSwCreationMode, isSwitchEnd)) {
+                createFictitiousSwitch(t, vl);
+            } else {
+                createInternalConnection(t, vl);
+            }
+        }
+
+        return cgmes2iidm.get(t.id());
+    }
+
+    private void createInternalConnectionsIfNeeded(String connectivityNode, VoltageLevel vl) {
+        List<CgmesTerminal> connectivityNodeTerminals = context.terminalMapping().getConnectivityNodeTerminals(connectivityNode);
+        switch (connectivityNodeTerminals.size()) {
+            case 0 -> throw new PowsyblException("Erroneous connectivity node terminals mapping");
+            case 1 ->
+                // Only one terminal: no need to create an internal connection
+                // Connectivity node and terminal share the same iidm node
+                oneIidmNodeForBothTerminalAndConnectivityNode(connectivityNodeTerminals.get(0), vl);
+            case 2 -> {
+                // There are two terminals on the same connectivity node. We need one (and only one!) internal connection
+                // between the two terminals as iidm can only handle one terminal per node.
+                // If there's one bbs, it is placed at the connectivity node: in iidm we want the node of the bbs terminal
+                // (and not the connectivity point!) to be where all feeders connect.
+                // If there are two bbs on the same connectivity node (that is, two united bbs), this is only done for
+                // the first bbs of the list.
+                CgmesTerminal t0 = connectivityNodeTerminals.get(0);
+                CgmesTerminal t1 = connectivityNodeTerminals.get(1);
+                if (!isBusbarSectionTerminal(t0) && isBusbarSectionTerminal(t1)) {
+                    oneIidmNodeForBothTerminalAndConnectivityNode(t1, vl);
+                    createInternalConnection(t0, vl);
+                } else {
+                    oneIidmNodeForBothTerminalAndConnectivityNode(t0, vl);
+                    createInternalConnection(t1, vl);
+                }
+            }
+            default -> {
```

Comment:
Can you add a unit test with more than 2 terminals at one ConnectivityNode?

Reply:
Unit test added covering all cases.